test/src/596-catalogbalancer/main

```diff
@@ -0,0 +1,108 @@
+cvmfs_test_name="Catalog balancer"
+cvmfs_test_autofs_on_startup=false
+
+add_files() {
+  local repo_dir=$1
+  local num_files=$2
+  mkdir -p "$repo_dir"
+  local body="$repo_dir/file"
+
+  for i in $(seq 1 1 $num_files); do
+    touch "$body$i"
+  done
+}
+
+add_catalog() {
+  local repo_dir=$1
+  touch "$repo_dir/.cvmfscatalog"
+}
+
+cvmfs_run_test() {
+  apache_switch on
+  local repo="test.local"
+  local repo_dir="/cvmfs/$repo"
+
+  create_repo "$repo" $(whoami)
+
+    sudo bash -c "echo 'CVMFS_AUTOCATALOG=true
+  CVMFS_AUTOCATALOG_MAX_WEIGHT=20
+  CVMFS_AUTOCATALOG_MIN_WEIGHT=8' >> /etc/cvmfs/repositories.d/$repo/server.conf"
+
+  start_transaction "$repo"
+  add_files "$repo_dir" 25
+  add_files "$repo_dir/dir100" 100
+  add_files "$repo_dir/dir3" 3
+  add_files "$repo_dir/dir4" 4
+  add_catalog "$repo_dir/dir4"
+  publish_repo "$repo"
+
+  echo "Checking first iteration"
```

Comment:
Please also use [`check_catalog_presence()`](https://github.com/cvmfs/cvmfs/blob/devel/test/test_functions#L768) to verify that the catalogs are actually created AND/OR not created.